posts/templates/posts/edit.html

```diff
@@ -3,8 +3,24 @@
 {% block content %}
 
     <h2 class="alert alert-warning">Edit this post here</h2>
-    <form action="{% url 'posts:edit' post.id post.skey %}" method="post" class="col-lg-8 mb-5 needs-validation">
+    {% if request.user.is_authenticated and  not post.creator %}
+            <button class="btn btn-lg btn-block btn-warning mb-3" onclick="return myFunction()">Claim this Post as yours</button>
+        <script>
+            function myFunction() {
+                var txt;
+                var r = confirm("Are you Sure you want to claim this Post? \n press OK if Yes");
+                if (r == true) {
+                    txt = "{{ request.user }}";
+                    window.location.href = "{% url 'posts:claim' post.id  post.skey %}";
```

Comment:
It should be a post request so make it a form submit. Things that change database state or session state should be a post request. There are a few exceptions